### PR TITLE
use a sink for file write

### PR DESCRIPTION
**Pull request by pq**
/cc @bwilkerson

### CODE REVIEW

File: example/widget_surveyor.dart [hunk 1]
```diff
@@ -160,6 +160,8 @@ class AnalysisResult {
       {'name': appName, 'widgets': widgetReferences};
 }
 
+// bug: fixed in linter 0.1.116 (remove once landed)
+// ignore: prefer_mixin
```
Comment (bwilkerson) [inline example/widget_surveyor.dart:164]: Was this moved by the sort feature? If so, we really need to fix it to keep preceding comments with a declaration.

Comment (pq) [reply example/widget_surveyor.dart:164]: Yep.  Sorting and formatting both wreak havoc on non-doc comments.

Comment (bwilkerson) [reply example/widget_surveyor.dart:164]: I'm not aware of any problems with formatting, only with sorting. If you can reproduce this we should let Bob know.

Comment (pq) [reply example/widget_surveyor.dart:164]: Whoops, I misspoke. I think the formatter is OK.  Sorry for the noise!

**Pull request merged**